### DISCUSSION TITLE
Cannot Install

### DISCUSSION THREAD

**Discussion by AlexThePotter** [syssi/esphome-jk-bms#146] [Q&A]
@syssi I can't seem to install your ble wifi esp32 example.

From HA esphome, I put your example yaml in, have my passwords in the secrets. 

If i used framework esp I get build error, if I do arduino I get boot loops?

Using esp32 devkit.

Please help this looks like a great help

**Comment by syssi** [comment]
Please provide the build error. I assume your HA instance lives on a Raspberry Pi?

**Reply by AlexThePotter** [reply]
Yes pi 4.

Error of build is:
Error: Could not find the package with 'platformio/toolchain-esp32ulp @ ~1.22851.0' requirements for your system 'linux_aarch64'

**Reply by syssi** [reply]
The ESP-IDF framework isn't aarch64 compatible at the moment. Please use a amd64 machine (linux/windows) to build & uplodate the configuration manually:

```
pip3 install esphome
esphome run esp32-ble-example.yaml
```

**Reply by AlexThePotter** [reply]
Right okay, will do.

I see the comments that Mqtt is only supported in the Arduino framework. Does this mean mqtt doesn't work for esp idf atm?

**Reply by syssi** [reply]
This isn't true anymore. MQTT at ESP-IDF is supported since a few ESPHome releases.
